aws/rust-runtime/aws-auth/src/provider/lazy_caching.rs

```diff
@@ -202,8 +199,10 @@ pub mod builder {
             );
             LazyCachingCredentialsProvider::new(
                 SystemTimeSource,
-                self.sleep.expect("sleep_impl is required"),
-                self.loader.expect("loader is required"),
+                self.sleep.unwrap_or_else(|| {
```

Comment:
Doesn't this fall under the category of programmer error? Or are you thinking there will be use-cases where `sleep` is set programmatically instead of hardcoded?